grails-app/services/org/pih/warehouse/order/OrderSummaryService.groovy

```diff
@@ -0,0 +1,441 @@
+                                    LEFT OUTER JOIN (
+                                        SELECT oi.order_id AS order_id,
+                                            SUM(oi.quantity * oi.unit_price) AS order_total
+                                        FROM order_item oi
+                                            LEFT OUTER JOIN `order` o ON o.id = oi.order_id
+                                        GROUP BY oi.order_id
+                                    ) AS order_total ON `order`.id = order_total.order_id
+                                WHERE order_adjustment.canceled IS NOT TRUE
+                                GROUP BY order_adjustment_id, order_id
+                            ) AS amount_per_order_by_adjustment
+                        GROUP BY amount_per_order_by_adjustment.order_id) AS total_adjustments ON total_adjustments.order_id = `order`.id
+                    WHERE `order`.order_type_id = 'PURCHASE_ORDER'
+                        AND `order`.id = '${orderId}'
+                        AND order_adjustment.canceled IS NOT TRUE
+                    GROUP BY `order`.id) AS items_and_adjustments_union
+                GROUP BY id, order_status) AS order_summary
+            WHERE id = '${orderId}'
+        """
+    }
+
+    /**
+     * Refreshing entire Order Summary materialized view, should be only triggered from time to time
+     * (same statements as in the order-summary-materialized-view.sql)
+     * */
+    def refreshOrderSummary() {
+        List statements = [
+            // Drop mv temp table if somehow it still exists
+            "DROP TABLE IF EXISTS order_summary_mv_temp;",
+            // Create temp mv table from sql view (to shorten the time when MV is unavailable)
+            "CREATE TABLE order_summary_mv_temp AS SELECT DISTINCT * FROM order_summary;",
+            "DROP TABLE IF EXISTS order_summary_mv;",
+            // Copy data from temp mv table into mv table
+            "CREATE TABLE IF NOT EXISTS order_summary_mv LIKE order_summary_mv_temp;",
+            "TRUNCATE order_summary_mv;",
+            "INSERT INTO order_summary_mv SELECT * FROM order_summary_mv_temp;",
+            "ALTER TABLE order_summary_mv ADD UNIQUE INDEX (id);",
+            // Cleanup
+            "DROP TABLE IF EXISTS order_summary_mv_temp;",
+        ]
+        dataService.executeStatements(statements)
+    }
+
+    /**
+     * Refreshing the Order Summary materialized view for a specific list of Order Ids (PASS ONLY A PO IDs)
+     * */
+    def refreshOrderSummary(List<String> orderIds, Boolean isDelete) {
+        List statements = []
+        orderIds?.each { String orderId ->
+            if (isDelete) {
+                statements << "DELETE FROM order_summary_mv WHERE id = '${orderId}';"
+            } else {
+                String orderSummarySelect = getOrderSummarySelect(orderId)
+                statements << "REPLACE INTO order_summary_mv (${orderSummarySelect});"
+            }
+        }
```

Comment:
you can write this function without pushing strings to the `statements`, you can use `inject`.

Reply:
I just moved this one